API Wrapper update

mkaizen opened the pull request:
Updated the API related files to use the api wrapper and changed the format to fit the different JSON response

mkaizen committed:
API Wrapper update

Updated the API related files to use the api wrapper and changed the format to fit the different JSON response

mkaizen committed:
Fixed images loading

Had old field value for product images on react vs api response they are now the same

mkaizen committed:
Updated itemGrid.tsx

Updated itemGrid to correctly display product cards in a 3 wide row of cards

qkstngus000 commented on open-recipes/src/api/OR_Api.tsx:
```diff
@@ -0,0 +1,28 @@
+  const postalCode = location ? await getPostalCodeFromLocation(location) : '98225';
+  const url = `${BASE_URL}?searchText=${query}&postalCode=${postalCode}`;
+  
+  const response = await fetch(url);
```
Should we use fetch or axios? Cuz I think axios is more compatible to current tech stack than fetch

mkaizen replied:
your right its probably better to switch to axios I can convert it

qkstngus000 commented on open-recipes/src/api/OR_Api.tsx:
IF this only contains wrapper functions and not react component, it should be ts file right?

qkstngus000 commented on open-recipes/src/app/index.tsx:
```diff
@@ -0,0 +1,149 @@
+
+export default function Page() {
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchResults, setSearchResults] = useState<FlippItem[]>([]);
```
I think if we are going to wrap the api call, we should not use any word like 'FXXXX' in anywhere in our code as well i think

qkstngus000 commented on open-recipes/src/components/ItemGrid.tsx:
```diff
@@ -0,0 +1,109 @@
+const cardsPerRow = 3;
+const cardWidth = (screenWidth - cardMargin * (cardsPerRow + 1)) / cardsPerRow;
+
+const ItemGrid: FC<ItemGridProps> = ({ items, isLoading, refreshing, onRefresh }) => {
```
I looked up what React FC is, and it is not recommended to use FC anymore and it will be out of practice and be deprecated in react 18.
https://medium.com/raccoons-group/why-you-probably-shouldnt-use-react-fc-to-type-your-react-components-37ca1243dd13

qkstngus000 replied:
Also https://github.com/payloadcms/payload/discussions/7513

qkstngus000 reviewed (changes requested):
I have left a reviews for each file i have questions with if you can review as well

mkaizen committed:
Michael review update

-Removed refrences to FXXX replaced with OR (Open recipes)
-Changed file type of Item.tsx to Item.ts and OR_Api.tsx to OR_Api.ts
-Changed Item Grid component to not use FC and explicitly define prop types

mkaizen committed:
fetch -> axios

- Added axios dependency
- switched usage of fetch to axios.get(url) in OR_Api.ts

matthew-latanafrancia commented on open-recipes/package.json:
```diff
@@ -17,16 +17,19 @@
     "expo-system-ui": "~3.0.7",
     "expo-web-browser": "~13.0.3",
     "nativewind": "^4.1.23",
+    "open-recipes": "file:",
```
is this a react thing? i'm not sure if this line needs to be added here.  adding a "file:" in the dependencies basically causes a circular dependency which is bad practice.

we might need to `rm -rf node_modules package-lock.json` on your local and re run `npm install` in `cd open-recipes` if this auto generated.  not sure if this is related but found this https://stackoverflow.com/questions/78887876/package-json-automatically-creating-default-dependencies-as-project-name-f

matthew-latanafrancia commented on open-recipes/src/api/OR_Api.ts:
file names in react native should be using `kebab-case` per react native naming practices https://www.restack.io/p/file-naming-best-practices-answer-react-native

so probably rename to something like `or-api.ts` or `open-recipes-api.ts` if we want to be more descriptive

matthew-latanafrancia replied:
to be honest i think i prefer PascalCase for components and camelCase for utility files so if we're ok with that I'm down with that too.  I can't really find any good sources for file name conventions for react but React is PascalCase and camelCase as I explained.

matthew-latanafrancia commented on open-recipes/src/api/OR_Api.ts:
```diff
@@ -0,0 +1,31 @@
+const getPostalCodeFromLocation = async (location: UserLocation): Promise<string> => {
+  try {
+    const response = await axios.get(
+      `https://nominatim.openstreetmap.org/reverse?format=json&lat=${location.latitude}&lon=${location.longitude}`
```
oooh this is cool

matthew-latanafrancia commented on open-recipes/src/api/OR_Api.ts:
```diff
@@ -0,0 +1,31 @@
+    return data.address.postcode || '98225';
+  } catch (error) {
+    console.error('Error getting postal code:', error);
+    return '98225';
```
can we add a todo comment here to change it from being a default zip code to just ask the user to input zip code?

matthew-latanafrancia commented on open-recipes/src/app/index.tsx:
```diff
@@ -0,0 +1,149 @@
+  }, []);
+
+  const getUserLocation = async () => {
+    setIsLocationLoading(true);
```
More things will definitely be added to this file in the future so it is preferred to not clutter this file when we can avoid it.   Maybe some things can be moved to utility files?  Like one specific for location stuffs, one for searches, etc.?

matthew-latanafrancia commented on open-recipes/src/app/index.tsx:
```diff
@@ -0,0 +1,149 @@
+      console.error('Error getting location:', error);
+      Alert.alert(
+        'Location Error',
+        'Unable to get your location. Default location will be used.',
```
or maybe the todo can be added here lol

matthew-latanafrancia commented on open-recipes/src/types/OR_Item.ts:
```diff
@@ -0,0 +1,17 @@
+export interface OR_Item {
```
```suggestion
export interface OpenRecipeServiceItem {
```

Would this be better for naming conventions?  OR i feel isn't that descriptive but lmk if I'm being too much lol

matthew-latanafrancia reviewed (commented):
left some comments.  file naming stuff can probably be left for discussion on tuesday.  better to think of something now and then stick with it

mkaizen committed:
Update open-recipes/src/types/OR_Item.ts

Co-authored-by: matthew-latanafrancia <77024792+matthew-latanafrancia@users.noreply.github.com>
